sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -456,9 +474,52 @@ case class Cast(child: Expression, dataType: DataType, timeZoneId: Option[String
     case DateType =>
       buildCast[Int](_, d => null)
     case TimestampType =>
-      buildCast[Long](_, t => timestampToLong(t).toInt)
-    case x: NumericType =>
-      b => x.numeric.asInstanceOf[Numeric[Any]].toInt(b)
+      buildCast[Long](_, t => {
+        val longValue = timestampToLong(t)
+        if (longValue == longValue.toInt) {
```

Comment:
no chance to has the same value?

Reply:
Do you mean: is it possible that`longValue == longValue.toInt` when longValue can't fit into `Int`? 
The `longValue` is long type, so it is impossible.

Reply:
oh, yeah. I missed it. Thanks!